python/dgl/nn/pytorch/gt/lap_enc.py

```diff
@@ -0,0 +1,160 @@
+    >>> from dgl import LaplacianPE
+    >>> from dgl.nn import LapPosEncoder
+
+    >>> transform = LaplacianPE(k=5, feat_name='eigvec', eigval_name='eigval', padding=True)
+    >>> g = dgl.graph(([0,1,2,3,4,2,3,1,4,0], [2,3,1,4,0,0,1,2,3,4]))
+    >>> g = transform(g)
+    >>> eig_vals, eig_vecs = g.ndata['eigval'], g.ndata['eigvec']
+    >>> TransformerLPE = LapPosEncoder(model_type="Transformer", num_layer=3, k=5,
+                                       lpe_dim=16, n_head=4)
+    >>> pos_enc = TransformerLPE(eig_vals, eig_vecs)
+    >>> DeepSetLPE = LapPosEncoder(model_type="DeepSet", num_layer=3, k=5,
+                                   lpe_dim=16, num_post_layer=2)
+    >>> pos_enc = DeepSetLPE(eig_vals, eig_vecs)
+    """
+
+    def __init__(
+        self,
+        model_type,
+        num_layer,
+        k,
+        lpe_dim,
+        n_head=1,
+        batch_norm=False,
+        num_post_layer=0,
+    ):
+        super(LapPosEncoder, self).__init__()
+        self.model_type = model_type
+        self.linear = nn.Linear(2, lpe_dim)
+
+        if self.model_type == "Transformer":
+            encoder_layer = nn.TransformerEncoderLayer(
+                d_model=lpe_dim, nhead=n_head, batch_first=True
+            )
+            self.pe_encoder = nn.TransformerEncoder(
+                encoder_layer, num_layers=num_layer
+            )
+        elif self.model_type == "DeepSet":
+            layers = []
+            if num_layer == 1:
+                layers.append(nn.ReLU())
+            else:
+                self.linear = nn.Linear(2, 2 * lpe_dim)
+                layers.append(nn.ReLU())
+                for _ in range(num_layer - 2):
+                    layers.append(nn.Linear(2 * lpe_dim, 2 * lpe_dim))
+                    layers.append(nn.ReLU())
+                layers.append(nn.Linear(2 * lpe_dim, lpe_dim))
+                layers.append(nn.ReLU())
+            self.pe_encoder = nn.Sequential(*layers)
+        else:
+            raise ValueError(
+                f"model_type '{model_type}' is not allowed, must be "
+                "'Transformer' or 'DeepSet'."
+            )
+
+        if batch_norm:
+            self.raw_norm = nn.BatchNorm1d(k)
+        else:
+            self.raw_norm = None
+
+        if num_post_layer > 0:
+            layers = []
+            if num_post_layer == 1:
+                layers.append(nn.Linear(lpe_dim, lpe_dim))
+                layers.append(nn.ReLU())
+            else:
+                layers.append(nn.Linear(lpe_dim, 2 * lpe_dim))
+                layers.append(nn.ReLU())
+                for _ in range(num_post_layer - 2):
+                    layers.append(nn.Linear(2 * lpe_dim, 2 * lpe_dim))
+                    layers.append(nn.ReLU())
+                layers.append(nn.Linear(2 * lpe_dim, lpe_dim))
+                layers.append(nn.ReLU())
+            self.post_mlp = nn.Sequential(*layers)
+        else:
+            self.post_mlp = None
+
+    def forward(self, eig_vals, eig_vecs):
```

Comment:
eigvals, eigvecs

eigenvalue and eigenvector are single word instead 2 words, not underscore required.